cloud/scope/managed_control_plane.go

```diff
@@ -282,7 +282,12 @@ func (s *ManagedControlPlaneScope) isResourceCreatedByClusterAPI(resourceFreeFor
 
 // GetClusterName returns the name of the cluster
 func (s *ManagedControlPlaneScope) GetClusterName() string {
-	return s.OCIManagedControlPlane.Name
+	okeClusterName := s.OCIManagedControlPlane.Name
+	// oke cluster names are restricted to less than 32
+	if len(okeClusterName) > 31 {
```

Comment:
Should we log that we are doing this? or at least document it in the book?

Also should our admission webhook do this?

Reply:
good call out, moved it to webhook